packages/ui/src/components/Visualization/Canvas/FormV2/hooks/one-of-field.ts

```diff
@@ -25,27 +25,31 @@ export const useOneOfField = (propName: string) => {
       return;
     }
 
-    if (typeof value === 'object' && isDefined(selectedOneOfSchema?.schema.properties)) {
-      const newValue = { ...value };
-      Object.keys(selectedOneOfSchema.schema.properties).forEach((prop) => {
-        delete (newValue as Record<string, unknown>)[prop];
-      });
-      onChange(newValue);
+    let newValue = {};
```

Comment:
This should be the same type of the `selectedOneOfSchema`, meaning that if the new value is object, we initialize it with `{}`, if `string` we initialize it with `''`, `number` with `0`.